# Find YY1 motifs in the P. picta genome

Based on the results from the homer's motif enrichment analysis the YY1 motif is variable at two positions with giving a total of 6 potential YY1 motifs.

The following code will search for each of these motifs independently, in forward and reverse orientation, and concatenate the results.

To search for the motif in other species, replace the genome file and chromosome names accordingly.

```{R}
library(Biostrings)

picta<-readDNAStringSet("picta_genome.fasta")
Chrs<-names(picta[1:23])
YY1pos<-lapply(Chrs,function(x) {
  RES<-matchPattern("CAGCCATTTTGT",picta[[x]])
  RES<-cbind(rep(x,length(RES)),RES@ranges@start)
  return(RES)
})
YY1pos<-do.call("rbind",YY1pos)
dim(YY1pos)
YY1startend<-cbind(YY1pos,as.numeric(YY1pos[,2])+1) #if needed
YY1startend<-as.data.frame(YY1startend,stringsAsFactors=F) #if needed
YY1startend[,2]<-as.numeric(YY1startend[,2]) #if needed
YY1startend[,3]<-as.numeric(YY1startend[,3]) #if needed
yy1.1<-YY1startend
nrow(yy1.1)

YY1pos<-lapply(Chrs,function(x) {
  RES<-matchPattern("CGGCCATTTTGT",picta[[x]])
  RES<-cbind(rep(x,length(RES)),RES@ranges@start)
  return(RES)
})
YY1pos<-do.call("rbind",YY1pos)
dim(YY1pos)
YY1startend<-cbind(YY1pos,as.numeric(YY1pos[,2])+1) #if needed
YY1startend<-as.data.frame(YY1startend,stringsAsFactors=F) #if needed
YY1startend[,2]<-as.numeric(YY1startend[,2]) #if needed
YY1startend[,3]<-as.numeric(YY1startend[,3]) #if needed
yy1.2<-YY1startend
nrow(yy1.2)

YY1pos<-lapply(Chrs,function(x) {
  RES<-matchPattern("CCGCCATATTGT",picta[[x]])
  RES<-cbind(rep(x,length(RES)),RES@ranges@start)
  return(RES)
})
YY1pos<-do.call("rbind",YY1pos)
dim(YY1pos)
YY1startend<-cbind(YY1pos,as.numeric(YY1pos[,2])+1) #if needed
YY1startend<-as.data.frame(YY1startend,stringsAsFactors=F) #if needed
YY1startend[,2]<-as.numeric(YY1startend[,2]) #if needed
YY1startend[,3]<-as.numeric(YY1startend[,3]) #if needed
yy1.3<-YY1startend
nrow(yy1.3)

YY1pos<-lapply(Chrs,function(x) {
  RES<-matchPattern("CCGCCATTTTGT",picta[[x]])
  RES<-cbind(rep(x,length(RES)),RES@ranges@start)
  return(RES)
})
YY1pos<-do.call("rbind",YY1pos)
dim(YY1pos)
YY1startend<-cbind(YY1pos,as.numeric(YY1pos[,2])+1) #if needed
YY1startend<-as.data.frame(YY1startend,stringsAsFactors=F) #if needed
YY1startend[,2]<-as.numeric(YY1startend[,2]) #if needed
YY1startend[,3]<-as.numeric(YY1startend[,3]) #if needed
yy1.4<-YY1startend
nrow(yy1.4)

YY1pos<-lapply(Chrs,function(x) {
  RES<-matchPattern("CAGCCATATTGT",picta[[x]])
  RES<-cbind(rep(x,length(RES)),RES@ranges@start)
  return(RES)
})
YY1pos<-do.call("rbind",YY1pos)
dim(YY1pos)
YY1startend<-cbind(YY1pos,as.numeric(YY1pos[,2])+1) #if needed
YY1startend<-as.data.frame(YY1startend,stringsAsFactors=F) #if needed
YY1startend[,2]<-as.numeric(YY1startend[,2]) #if needed
YY1startend[,3]<-as.numeric(YY1startend[,3]) #if needed
yy1.5<-YY1startend
nrow(yy1.5)

YY1pos<-lapply(Chrs,function(x) {
  RES<-matchPattern("CGGCCATATTGT",picta[[x]])
  RES<-cbind(rep(x,length(RES)),RES@ranges@start)
  return(RES)
})
YY1pos<-do.call("rbind",YY1pos)
dim(YY1pos)
YY1startend<-cbind(YY1pos,as.numeric(YY1pos[,2])+1) #if needed
YY1startend<-as.data.frame(YY1startend,stringsAsFactors=F) #if needed
YY1startend[,2]<-as.numeric(YY1startend[,2]) #if needed
YY1startend[,3]<-as.numeric(YY1startend[,3]) #if needed
yy1.6<-YY1startend
nrow(yy1.6)


yy1.1$V1<-factor(yy1.1$V1,levels=Chrs)

yy1.2$V1<-factor(yy1.2$V1,levels=Chrs)

yy1.3$V1<-factor(yy1.3$V1,levels=Chrs)

yy1.4$V1<-factor(yy1.4$V1,levels=Chrs)

yy1.5$V1<-factor(yy1.5$V1,levels=Chrs)

yy1.6$V1<-factor(yy1.6$V1,levels=Chrs)

yy1.1.picta<-as.data.frame(table(yy1.1$V1))
yy1.2.picta<-as.data.frame(table(yy1.2$V1))
yy1.3.picta<-as.data.frame(table(yy1.3$V1))
yy1.4.picta<-as.data.frame(table(yy1.4$V1))
yy1.5.picta<-as.data.frame(table(yy1.5$V1))
yy1.6.picta<-as.data.frame(table(yy1.6$V1))



####################Reverse Complement

YY1pos<-lapply(Chrs,function(x) {
  RES<-matchPattern("ACAAAATGGCTG",picta[[x]])
  RES<-cbind(rep(x,length(RES)),RES@ranges@start)
  return(RES)
})
YY1pos<-do.call("rbind",YY1pos)
dim(YY1pos)
YY1startend<-cbind(YY1pos,as.numeric(YY1pos[,2])+1) #if needed
YY1startend<-as.data.frame(YY1startend,stringsAsFactors=F) #if needed
YY1startend[,2]<-as.numeric(YY1startend[,2]) #if needed
YY1startend[,3]<-as.numeric(YY1startend[,3]) #if needed
yy1.1.rev<-YY1startend
nrow(yy1.1.rev)

YY1pos<-lapply(Chrs,function(x) {
  RES<-matchPattern("ACAAAATGGCCG",picta[[x]])
  RES<-cbind(rep(x,length(RES)),RES@ranges@start)
  return(RES)
})
YY1pos<-do.call("rbind",YY1pos)
dim(YY1pos)
YY1startend<-cbind(YY1pos,as.numeric(YY1pos[,2])+1) #if needed
YY1startend<-as.data.frame(YY1startend,stringsAsFactors=F) #if needed
YY1startend[,2]<-as.numeric(YY1startend[,2]) #if needed
YY1startend[,3]<-as.numeric(YY1startend[,3]) #if needed
yy1.2.rev<-YY1startend
nrow(yy1.2.rev)

YY1pos<-lapply(Chrs,function(x) {
  RES<-matchPattern("ACAATATGGCGG",picta[[x]])
  RES<-cbind(rep(x,length(RES)),RES@ranges@start)
  return(RES)
})
YY1pos<-do.call("rbind",YY1pos)
dim(YY1pos)
YY1startend<-cbind(YY1pos,as.numeric(YY1pos[,2])+1) #if needed
YY1startend<-as.data.frame(YY1startend,stringsAsFactors=F) #if needed
YY1startend[,2]<-as.numeric(YY1startend[,2]) #if needed
YY1startend[,3]<-as.numeric(YY1startend[,3]) #if needed
yy1.3.rev<-YY1startend
nrow(yy1.3.rev)

YY1pos<-lapply(Chrs,function(x) {
  RES<-matchPattern("ACAAAATGGCGG",picta[[x]])
  RES<-cbind(rep(x,length(RES)),RES@ranges@start)
  return(RES)
})
YY1pos<-do.call("rbind",YY1pos)
dim(YY1pos)
YY1startend<-cbind(YY1pos,as.numeric(YY1pos[,2])+1) #if needed
YY1startend<-as.data.frame(YY1startend,stringsAsFactors=F) #if needed
YY1startend[,2]<-as.numeric(YY1startend[,2]) #if needed
YY1startend[,3]<-as.numeric(YY1startend[,3]) #if needed
yy1.4.rev<-YY1startend
nrow(yy1.4.rev)

YY1pos<-lapply(Chrs,function(x) {
  RES<-matchPattern("ACAATATGGCTG",picta[[x]])
  RES<-cbind(rep(x,length(RES)),RES@ranges@start)
  return(RES)
})
YY1pos<-do.call("rbind",YY1pos)
dim(YY1pos)
YY1startend<-cbind(YY1pos,as.numeric(YY1pos[,2])+1) #if needed
YY1startend<-as.data.frame(YY1startend,stringsAsFactors=F) #if needed
YY1startend[,2]<-as.numeric(YY1startend[,2]) #if needed
YY1startend[,3]<-as.numeric(YY1startend[,3]) #if needed
yy1.5.rev<-YY1startend
nrow(yy1.5.rev)

YY1pos<-lapply(Chrs,function(x) {
  RES<-matchPattern("ACAATATGGCCG",picta[[x]])
  RES<-cbind(rep(x,length(RES)),RES@ranges@start)
  return(RES)
})
YY1pos<-do.call("rbind",YY1pos)
dim(YY1pos)
YY1startend<-cbind(YY1pos,as.numeric(YY1pos[,2])+1) #if needed
YY1startend<-as.data.frame(YY1startend,stringsAsFactors=F) #if needed
YY1startend[,2]<-as.numeric(YY1startend[,2]) #if needed
YY1startend[,3]<-as.numeric(YY1startend[,3]) #if needed
yy1.6.rev<-YY1startend
nrow(yy1.6.rev)


Chrs<-paste0("Scaffold_",c("1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23"),sep="")

yy1.all<-rbind(yy1.1,yy1.2,yy1.3,yy1.4,yy1.5,yy1.6,yy1.1.rev,yy1.2.rev,yy1.3.rev,yy1.4.rev,yy1.5.rev,yy1.6.rev)
yy1.all$V1<-factor(yy1.all$V1,levels=Chrs)
yy1.all$V3<-yy1.all$V2+11
colnames(yy1.all)<-c("chr","start","end")
write.csv(yy1.all,file="yy1_locations.csv",row.names=FALSE,quote=FALSE)
yy1.table<-as.data.frame(table(yy1.all$V1))




yy1.1.rev$V1<-factor(yy1.1.rev$V1,levels=Chrs)

yy1.2.rev$V1<-factor(yy1.2.rev$V1,levels=Chrs)

yy1.3.rev$V1<-factor(yy1.3.rev$V1,levels=Chrs)

yy1.4.rev$V1<-factor(yy1.4.rev$V1,levels=Chrs)

yy1.5.rev$V1<-factor(yy1.5.rev$V1,levels=Chrs)

yy1.6.rev$V1<-factor(yy1.6.rev$V1,levels=Chrs)


yy1.1<-as.data.frame(table(yy1.1.rev$V1))
yy1.2<-as.data.frame(table(yy1.2.rev$V1))
yy1.3<-as.data.frame(table(yy1.3.rev$V1))
yy1.4<-as.data.frame(table(yy1.4.rev$V1))
yy1.5<-as.data.frame(table(yy1.5.rev$V1))
yy1.6<-as.data.frame(table(yy1.6.rev$V1))

yy1.1<-cbind(yy1.1.picta,yy1.1[,2])
yy1.2<-cbind(yy1.2.picta,yy1.2[,2])
yy1.3<-cbind(yy1.3.picta,yy1.3[,2])
yy1.4<-cbind(yy1.4.picta,yy1.4[,2])
yy1.5<-cbind(yy1.5.picta,yy1.5[,2])
yy1.6<-cbind(yy1.6.picta,yy1.6[,2])

yy1.1$counts<-yy1.1[,2] + yy1.1[,3]
yy1.2$counts<-yy1.2[,2] + yy1.2[,3]
yy1.3$counts<-yy1.3[,2] + yy1.3[,3]
yy1.4$counts<-yy1.4[,2] + yy1.4[,3]
yy1.5$counts<-yy1.5[,2] + yy1.5[,3]
yy1.6$counts<-yy1.6[,2] + yy1.6[,3]

yy1.1<-yy1.1[,c(1,4)]
yy1.2<-yy1.2[,c(1,4)]
yy1.3<-yy1.3[,c(1,4)]
yy1.4<-yy1.4[,c(1,4)]
yy1.5<-yy1.5[,c(1,4)]
yy1.6<-yy1.6[,c(1,4)]


yy1.picta<-as.matrix(cbind(yy1.1,yy1.2,yy1.3,yy1.4,yy1.5,yy1.6))
row.names(yy1.picta)<-c(1:23)
yy1.picta<-yy1.picta[,c(2,4,6,8,10,12)]
colnames(yy1.picta)<-c("CAGCCATTTTGT","CGGCCATTTTGT","CCGCCATATTGT","CCGCCATTTTGT","CAGCCATATTGT","CGGCCATATTGT")

yy1.picta<-(t(yy1.picta))

barplot(yy1.picta,
        col=c("midnightblue","slateblue4","skyblue4","lightsteelblue2","lavender","darkgoldenrod2"),
        border="white",
        space=0.04,
        font.axis=2,
        xlab="",
        ylim=c(0,500),
        las=1,
        main="Picta genome yy1 motif distribution")


```
